Ginger/GingerTest/APIModelLib/POMsTest.cs

```diff
@@ -0,0 +1,136 @@
+        }
+
+
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            GingerAutomator.EndSession();
+        }
+
+        [TestMethod]
+        public void ValidatePOMWasAddedToPOMsTree()
+        {
+            //Act
+            mPOMsPOM.SelectPOM(mLearnedPOM.Name);
+            ApplicationPOMModel treePOM = (from x in WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ApplicationPOMModel>() where x.Name == mLearnedPOM.Name select x).SingleOrDefault();
+
+            //Assert
+            Assert.AreEqual(mLearnedPOM.Name, treePOM.Name, "POM.Name is same");
+        }
+
+
+        [TestMethod]
+        public void ValidatePOMGeneralDetails()
+        {
+            //Assert
+            Assert.AreEqual(mLearnedPOM.Name, "MyNewPOM", "POM.Name check");
+            Assert.AreEqual(mLearnedPOM.Description, "MyDescription", "POM.Description check");
+            Assert.IsTrue(mLearnedPOM.PageURL.EndsWith("TestResources/HTML/HTMLControls.html"), "POM.URL check");
+            Assert.AreEqual(mLearnedPOM.TargetApplicationKey.ToString(), "MyWebApp~843fb8a6-a844-45e9-b7ef-b045b6433c44", "POM.TargetApplicationKey is same");
+        }
+
+
+        [TestMethod]
+        public void ValidatePOMScreenshotWasTaken()
+        {
+            //Act
+            BitmapSource source = Ginger.General.GetImageStream(Ginger.General.Base64StringToImage(mLearnedPOM.ScreenShotImage.ToString()));
+            //Assert  
+            Assert.IsNotNull(source, "POM.ScreenShotImage converted to sourse check");
+        }
+
+
+        [TestMethod]
+        public void ValidateLearnedItems()
+        {
+            //Act
+            ElementInfo EI1 = mLearnedPOM.MappedUIElements.Where(x => x.ElementName == "Mexico INPUT.RADIO" && x.ElementTypeEnum == eElementType.RadioButton).FirstOrDefault();
+            ElementInfo EI2 = mLearnedPOM.MappedUIElements.Where(x => x.ElementName == "id123 input" && x.ElementTypeEnum == eElementType.TextBox).FirstOrDefault();
+
+            //Assert  
+            Assert.AreEqual(mLearnedPOM.MappedUIElements.Count, 42, "POM.MappedUIElements.Coun check");
+            Assert.AreEqual(mLearnedPOM.UnMappedUIElements.Count, 77, "POM.UnMappedUIElements.Count check");
+            Assert.IsNotNull(EI1, "POM.Element learned check");
+            Assert.IsNotNull(EI2, "POM.Element learned check");
+        }
+
+        [TestMethod]
+        public void ValidateElementsProperties()
+        {
+            //Assert  
+            Assert.AreEqual(mLearnedPOM.MappedUIElements[0].Properties.Count, 8, "POM.properties check");
+            Assert.AreEqual(mLearnedPOM.UnMappedUIElements[1].Properties.Count, 7, "POM.properties check");
+        }
+
+
+        [TestMethod]
+        public void ValidateElementsLocators()
+        {
+            //Assert  
```

Comment:
not only count, pelease check full list on locators that we fin for 1 button and 1 drop list and validate the xPath, RleativeXpath, Id, name, all we have on those elements